src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -43,12 +42,12 @@ protected override ConversionsBase WithNullabilityCore(bool includeNullability)
         public override Conversion GetMethodGroupDelegateConversion(BoundMethodGroup source, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
         {
             // Must be a bona fide delegate type, not an expression tree type.
-            if (!destination.IsDelegateType())
+            if (!(destination.IsDelegateType() || destination.SpecialType == SpecialType.System_Delegate))
```

Comment:
Should we also allow `destination` to be `object`, or `ICloneable`, or `MulticastDelegate` here?

Reply:
> Should we also allow destination to be object, or ICloneable, or MulticastDelegate here?

Perhaps. Let's address later when handling those cases.